### PR TITLE
Cleanup/tidy fake-api-gateway

### PR DESCRIPTION
**Pull request by Raynos**
There's dead code and old APIs that predate our
current usage of FakeAPIGAteway in the functions
application.

This is a breaking change and fixes the fact we have
three different ways of "adding workers" and now
forcing everyone to use `addWorker()` 

Also deletes dead files, moves docker feature to
an experiment directory.

### CODE REVIEW

File: child-process-worker.js [hunk 1]
```diff
@@ -2,14 +2,19 @@
 'use strict'
 
 const childProcess = require('child_process')
-const util = require('./util')
+const fs = require('fs')
+const path = require('path')
 
-const WORKER_PATH = '/tmp/worker.js' // path.join(__dirname, 'worker.js')
+const WORKER_PATH = '/tmp/fake-api-gateway-lambda/worker.js'
```
Comment (jwerle) [inline child-process-worker.js:8]: Should we eventually use `os.tmpdir()` or https://github.com/sindresorhus/temp-dir ?

File: worker.js [hunk 2]
```diff
@@ -113,106 +97,110 @@ const main = module.exports = function () {
      * that we can borrow implementations from.
      */
 
-      const maybePromise = this.lambdaFunction.handler(eventObject, {}, (err, result) => {
-        if (!result) {
-          this.sendError(id, err)
-          return
-        }
+    console.error('this.handler field', this.handler)
```
Comment (jwerle) [inline worker.js:100]: should this be here?

Comment (Raynos) [reply worker.js:100]: This is a dead print i added for debugging, needs to be removed.

File: child-process-worker.js (hunk 1 above)
Comment (Raynos) [reply child-process-worker.js:8]: :+1: willfix.

**Pull request merged**